terraform-setup/analytics/redshift/main.tf

```diff
@@ -31,11 +31,16 @@ resource "aws_ssm_parameter" "redshift_admin_password"{
   value = random_password.redshift_password.result
 }
 
-# Namespace
+#Retrieve existing admin username created and saved in AWS SSM parameter store
+data "aws_ssm_parameter" "redshift_admin_username"{
+  name = "/redshift/admin_username"
```

Comment:
You need to distinguish the environment of the cluster, what if you have another cluster for staging? In my opinion, you can just have a paramter name like `/production/redshift/admin_username`